### ISSUE TITLE
Implement HUD Elements

### ISSUE THREAD

**Issue by AnthonyWharton** [CosyClub/ProjectTempo#25]
We need the ability to display different things in the HUD.

We will need:

- ~Health Bar (Top/Bottom) (currently above players, no big bar for boss or local player)~
- ~Combo Display (Bottom Left)~
- ~Beat Indicator (Left hand side of the screen)~
- Notification/Text across the upper-centre region of the screen.

**Comment by Codrin-Popa** [comment]
```
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/lib/libOgreOverlayStatic_d.a(compile_OgreOverlay_0.cpp.o): In function `Ogre::Font::loadResource(Ogre::Resource*)':
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/Components/Overlay/src/OgreFont.cpp:250: undefined reference to `FT_Init_FreeType'
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/Components/Overlay/src/OgreFont.cpp:264: undefined reference to `FT_New_Memory_Face'
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/Components/Overlay/src/OgreFont.cpp:271: undefined reference to `FT_Set_Char_Size'
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/Components/Overlay/src/OgreFont.cpp:293: undefined reference to `FT_Load_Char'
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/Components/Overlay/src/OgreFont.cpp:357: undefined reference to `FT_Load_Char'
/home/codrin/Desktop/ProjectTempo/extlibs/ogre-1.10.9/Components/Overlay/src/OgreFont.cpp:434: undefined reference to `FT_Done_FreeType'
collect2: error: ld returned 1 exit status
src/client/CMakeFiles/client.dir/build.make:110: recipe for target '../bin/RaveCave' failed
make[2]: *** [../bin/RaveCave] Error 1
CMakeFiles/Makefile2:723: recipe for target 'src/client/CMakeFiles/client.dir/all' failed
make[1]: *** [src/client/CMakeFiles/client.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```
We need freetype linked in order to use overlay for displaying text and the HUD

**Comment by Codrin-Popa** [comment]
Finally, we have some sample text and boxes displayed on the screen after 8-10h of struggle with Ogre

**Comment by grant-m-s** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [HUD/UI](https://trello.com/c/voUrazOR/5-hud-ui)

**Comment by grant-m-s** [comment]
Health bar still needs implementing

**Comment by Codrin-Popa** [comment]
Health bar implemented

**Comment by qqii** [comment]
It would be nice to have a little more control with the colors. I'd imagine some kind of mapping between percentage and a color, then interpolate the values between that. So right now the mapping is:

100% = Green
0% = Red

It instead becomes for example:

100% = Green
50% = Yellow
10% = Red
0% = Darker Red

It'll then interpolate between them, but we can be more selective about the colors.

**Comment by qqii** [comment]
In terms of the interpolation, I'd convert the colors into HSV and then interpolate in that space. It'll produce less murky colors.

**Comment by qqii** [comment]
Looks like someone's [made the code](http://irrlicht.sourceforge.net/forum/viewtopic.php?f=9&t=50102) in irrlicht too!

**Comment by qqii** [comment]
Waiting on #83 and related to #56.

**Comment by grant-m-s** [comment]
Only need #83 to be completed now. This can be closed.
